### DISCUSSION TITLE
Adding package to OpenSuSE Tumbleweed

### DISCUSSION THREAD

**Discussion by gato-mulatto** [facultyai/dash-bootstrap-components#1044] [Q&A] [answered]
Apologies for a very basic question.

How can I get dash-bootstrap-components package in OpenSuSe Tumbleweed with native python 3.11.9-2.1? 

I am working in OpenSuSe Tumbleweed with native python 3.11.9-2.1. The following dash packages are available and installed.

python311-dash 2.16.1-1.1
python311-dash-core-components 2.0.0-1.12
python311-dash-html-components 2.0.0-1.12
python311-dash-renderer 1.91-1.18
python311-dash-table 5.0.0-1.12

There is no dash-bootstrap-components package that I can see. Native environment should not updated by pip or conda.

Is there another way I can install the dash-bootstrap-components package in OpenSuSe Tumbleweed with native python 3.11.9-2.1?

Thank you for any help or clue.

Gato

**Comment by tcbegley** [comment]
I'm afraid I'm not really familiar with openSUSE Tumbleweed. Why do you want to use the system / native Python? Is it not an option to create a virtual environment and install packages there?

**Comment by gato-mulatto** [comment]
Perhaps I will resort to that, but doesn't feel natural given the native environment. Probably explore not using dash-bootstrap-components first if it is not widely available.

Any thoughts about how to install package natively ?

⁣

Michael Hieb
O:      +852.3112.3157
M:     +852.9461.3111
***@***.*** 

Bahlam Core Limited
20th Floor, Central Tower
28 Queen’s Road, Central
Hong Kong

Sent by BlueMail for Android. Please excuse my brevity and any typos.​

**Reply by tcbegley** [reply, marked as answer]
Virtual environments in Python are standard practice, and generally preferable to using a system installation of Python, since the operating system itself uses it and you don't want to introduce dependency conflicts. You can use a package manager to ensure no conflicts arise between installed packages, or you can sidestep the whole issue by using a virtual environment for each of your projects which is much simpler and generally recommended.

dash-bootstrap-components is widely available through both PyPI and conda-forge. We don't have the resources to maintain additional installation options for the many different operating specific package managers out there. Sorry if this means you don't feel you can use the library.

**Reply by gato-mulatto** [reply]
I have looked into it. Your advice is sound. It is best to run a virtual environment for each development project and leave the system install alone. I appreciate the advice.
